connectors/byoc.py

```diff
@@ -278,6 +316,8 @@ def get_filter(self, filter_state="active", domain=DEFAULT_DOMAIN):
 
 class PipelineSettings:
     def __init__(self, pipeline):
```

Comment:
`pipeline=None` ?

Reply:
To instantiate an instance of `PipelineSettings`, it's expected to provide a `pipeline`, so I am inclined to make it mandatory. But we couldn't stop someone using `pipeline = PipelineSettings(pipeline=None)`, therefore I added additional protection:
```python
if pipeline is None:
    pipeline = {}
```

Reply:
yes but this protection is just for None. This would break too:

```
pipeline = PipelineSettings(pipeline=1)
```

If pipeline is mandatory, you have to trust it's a pipeline. If there's no way to trust it, you can do:

```
if not isinstance(pipeline, dict):
    raise TypeError(pipeline)
```

but usually, we just trust it -- and you will get an error on the first attempt to use a dict method ("if it quacks like a duck, it's a duck")

Reply:
For this case, I think we do need `pipeline` in the `__init__` method. so I think we can keep it as 
```python
def __init__(self, pipeline):
```

Do you think it's unnecessary to have this?
```python
if pipeline is None:
    pipeline = {}
```